### PR TITLE
Add dashboard_script.py

### PR DESCRIPTION
**Pull request by apoorvaeternity**
This is the main script that will be run using the cron.
It creates a dataset_details.json file that stores the md5 and status of dataset.
I have used [FileLock](https://pypi.org/project/filelock/) for updating the dataset_details.json using multiple threads.

**The script is still under testing right now**

### CODE REVIEW

File: retrieverdash/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
-Django==2.0.2
+Django==1.11
+filelock
```
Comment (henrykironde) [inline retrieverdash/requirements.txt:2]: Why are we using 1.11?

Comment (apoorvaeternity) [reply retrieverdash/requirements.txt:2]: Django from version 2.0 has stopped supporting python 2.
So for supporting both python 2 and 3 we will have to stick to Django version 1.11

Comment (henrykironde) [reply retrieverdash/requirements.txt:2]: We don't have to support python 2 for the dashboard project.

**Comment by henrykironde** [comment]
if we are using that style of docs strings, I think we should include all parameters. 

```python
def create_diff(csv1, csv2, diff_file, context, numlines):
    """
    Parameters
    ----------
    csv1 : The first csv file.
    csv2 : The second csv file.
    diff_file : The diff_file that is to be generated.
    context: .....
    numlines: .....

```

File: retrieverdash/requirements.txt (hunk 1 above)
Comment (apoorvaeternity) [reply retrieverdash/requirements.txt:2]: Okay :+1:

File: retrieverdash/dashboard_script/dashboard_script.py [hunk 2]
```diff
@@ -30,8 +31,8 @@ def check_dataset(dataset):
     status = None
     try:
         md5 = get_dataset_md5(dataset)
-        if dataset.name not in dataset_detail or md5 != dataset_detail[
-                dataset.name]['md5']:
+        if (dataset.name not in dataset_detail
```
Comment (henrykironde) [inline retrieverdash/dashboard_script/dashboard_script.py]: remove the `()` they are redundant.

File: retrieverdash/dashboard_script/dashboard_script.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import json
+import os
+from multiprocessing import Pool
+
+from filelock import FileLock
+from retriever import datasets
+
+from status_dashboard_tools import (
```
Comment (henrykironde) [inline retrieverdash/dashboard_script/dashboard_script.py]: I really do not feel comfortable with this .🙂 let's change it to one line.

File: retrieverdash/dashboard_script/dashboard_script.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    get_dataset_md5,
+    diff_generator,
+    create_dirs,
+    dataset_to_csv)
+
+file_location = os.path.dirname(os.path.realpath(__file__))
+
+example_datasets = ['abalone-age', 'airports', 'biodiversity-response', 'bird-size',
```
Comment (henrykironde) [inline retrieverdash/dashboard_script/dashboard_script.py]: This is too long, P8P recommends nothing longer than 80, unless you compromise readability.

```python
example_datasets = [
    'abalone-age',
    'airports',
    'biodiversity-response',
    'bird-size',
    'breast-cancer-wi',
    'butterfly-population-network',
    'partners-in-flight',
    'portal']
``` 

or 

```
example_datasets = [
    'abalone-age', 'airports',
    'biodiversity-response', 'bird-size',
    'breast-cancer-wi', 'butterfly-population-network',
    'partners-in-flight', 'portal']
```

File: retrieverdash/dashboard_script/status_dashboard_tools.py [hunk 5]
```diff
@@ -55,6 +60,13 @@ def create_diff(csv1, csv2, diff_file, context, numlines):
     csv1 : The first csv file.
     csv2 : The second csv file.
     diff_file : The diff_file that is to be generated.
+    context : set to True for contextual differences (defaults to False
+            which shows full differences).
```
Comment (henrykironde) [inline retrieverdash/dashboard_script/status_dashboard_tools.py]: I would reword this 

```python
def create_diff(csv1, csv2, diff_file, context, numlines):
    """
    Parameters
    ----------
    csv1 : The first csv file.
    csv2 : The second csv file.
    diff_file : The diff_file that is to be generated.
    context : Defaults to false for full diff, set to True for contextual diff.
```

Comment (henrykironde) [reply retrieverdash/dashboard_script/status_dashboard_tools.py]: I feel the second part needs to be made more clear.
  `numlines : Number of context lines before and after diff.
    Set to True to controls number of code lines before and after the change.
    Set to False .....` I am having hard time to comprehend what `Set to False..` does

File: retrieverdash/dashboard_script/dashboard_script.py (hunk 3 above)
Comment (apoorvaeternity) [reply retrieverdash/dashboard_script/dashboard_script.py]: Hehe. Okay :smile:

File: retrieverdash/dashboard_script/status_dashboard_tools.py (hunk 5 above)
Comment (apoorvaeternity) [reply retrieverdash/dashboard_script/status_dashboard_tools.py]: Setting to false shows the whole file. The changes and also those lines which do not have changes.

Comment (apoorvaeternity) [reply retrieverdash/dashboard_script/status_dashboard_tools.py]: And when False is set for context the numlines are used for positioning the `next` anchor tag before the change. If context is set to false and numlines is set to 2 then it will place the anchor tag 2 lines before the change.

**Comment by henrykironde** [comment]
Use an ignore or example list as in https://github.com/weecology/retriever/blob/master/retriever/try_install_all.py#L77.
I recommend that you start with a small set of datasets that can actually run easily. The datasets like predicts, vertnets* take a long time. There some datasets like `bioclim, Mammal Super Tree, breed-bird-survey*` that cannot load into the databases on the current master branch.

If we can run  the code with minimal datasets, that will be good.
Later, we can add datasets to the list or remove.
